packages/bot-web-ui/src/pages/dashboard/announcements/__tests__/announcements.spec.tsx

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+import { mockStore, StoreProvider } from '@deriv/stores';
+import { render, screen, waitFor } from '@testing-library/react';
+import { mock_ws } from 'Utils/mock';
+import RootStore from 'Stores/index';
+import { DBotStoreProvider, mockDBotStore } from 'Stores/useDBotStore';
+import Announcements from '../announcements';
+import userEvent from '@testing-library/user-event';
+import { DBOT_TABS } from 'Constants/bot-contents';
+
+jest.mock('@deriv/bot-skeleton/src/scratch/dbot', () => jest.fn());
+jest.mock('@deriv/bot-skeleton/src/scratch/xml/main.xml', () => '<xml>sample</xml>');
+window.Blockly = {
+    utils: {
+        xml: {
+            textToDom: (xmlString: string) => {
+                const parser = new DOMParser();
+                return parser.parseFromString(xmlString, 'text/xml');
+            },
+        },
+    },
+    Xml: { domToText: () => ({}) },
+};
+
+const mockHandleTabChange = jest.fn();
+
+describe('Announcements', () => {
+    let wrapper: ({ children }: { children: JSX.Element }) => JSX.Element, mock_DBot_store: RootStore | undefined;
+    const mock_store = mockStore({});
+
+    beforeAll(() => {
+        mock_DBot_store = mockDBotStore(mock_store, mock_ws);
+
+        wrapper = ({ children }: { children: JSX.Element }) => (
+            <StoreProvider store={mock_store}>
+                <DBotStoreProvider ws={mock_ws} mock={mock_DBot_store}>
+                    {children}
+                </DBotStoreProvider>
+            </StoreProvider>
+        );
+    });
+
+    it('should decrease the indicator count and remove it, the list of announcements should be displayed, and redirect to tutorial page upon clicking on the announcement item.', async () => {
+        const { container } = render(<Announcements handleTabChange={mockHandleTabChange} is_mobile={true} />, {
+            wrapper,
+        });
+        const button = screen.getByTestId('btn-announcements');
+        userEvent.click(button);
+
+        await waitFor(() => {
+            // eslint-disable-next-line testing-library/no-node-access, testing-library/no-container
+            const notification_button = container.querySelector('.notification__button');
+
+            expect(notification_button).toBeInTheDocument();
+
+            if (notification_button) {
+                userEvent.click(notification_button);
+            }
+        })
+            .then(() => {
+                expect(screen.queryByTestId('announcements__amount')).not.toBeInTheDocument();
+
+                const button_cancel = screen.getByRole('button', { name: /Learn more/i });
+                userEvent.click(button_cancel);
+            })
+            .then(() => {
+                expect(mock_DBot_store?.dashboard.setActiveTab(DBOT_TABS.TUTORIAL));
+            });
```

Comment:
@maryia-matskevich-deriv Since we are using async await, is it necessary to go for the promise chain .then() approach? Shouldn't using `await` be more readable in this case?

Reply:
@shafin-deriv yes, we can